File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.go
func canMakeArithmeticProgression(arr []int) bool {
    if len(arr) <=2{
        return true
    }
        
    sort.Sort(sort.Reverse(sort.IntSlice(arr)))
    
    res_diff := arr[0] - arr[1]

    for i:=2;i<len(arr);i++{
        if arr[i-1] - arr[i] != res_diff{
            return false
        }
    }
    return true
}

File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]):
        if len(arr) <=2:
            return True
        
        arr.sort(reverse=True)
        res_diff = arr[0] - arr[1]
        
        for i in range(2, len(arr)):
            if arr[i-1] - arr[i] != res_diff:
                return False
        
        return True